custom/enikshay/case_utils.py

```diff
@@ -288,7 +273,7 @@ def get_episode_case_from_adherence(domain, adherence_case_id):
     Assumes the following case structure:
     Episode <--ext-- Adherence
     """
-    return get_parent_of_case(domain, adherence_case_id, CASE_TYPE_EPISODE)
+    return get_first_parent_of_case(domain, adherence_case_id, CASE_TYPE_EPISODE)
```

Comment:
`get_parent_of_case` gets open parent case and the `get_first_parent_of_case` gets first parent irrespective of status. 

I am not sure how safe a change that it. Its used at two places and i don't have context on either. If you have considered how it effects then its fine.

Reply:
```get_episode_case_from_adherence``` gets used in repeaters and ```get_prescription_from_voucher``` gets used in the reconciliation task - maybe @proteusvacuum can weigh in on those two uses?

Reply:
- For the task, it looks like we don't send an update if the prescription is closed. We could change the `try/except` there to a if `prescription.closed`.
- Similarly, in the `NinetyNineDotsAdherenceRepeater` we should check for a closed episode instead.